### ISSUE TITLE
Report (reduce str ...) as a problem

### ISSUE THREAD

**Issue by pmonks** [clj-kondo/clj-kondo#1734]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
I've been informed that (unusually), `(reduce str ...)` is problematic, because it creates a lot of garbage, and that either `(clojure.string/join ...)` or `(apply str ...)` are preferable.

**Describe the solution you'd like**
It would be great if clj-kondo flagged this, as it's the kind of unobvious "exception to the rule" that trips a lot of people up (newcomers especially).

**Describe alternatives you've considered**
I may also raise this with the eastwood project (the other linter I use from time to time), though that project seems less active than this one.

**Additional context**
Love your work!  😃

**Comment by borkdude** [comment]
I agree that this would be useful to warn on. Are there more cases where something is built up using a mutable accumulator? I.e. could the linter be generalized, or should we go with `:reduce-str` + `"Use (apply str ...) or (str/join ...)"` or so.

**Comment by pmonks** [comment]
Sorry I have no idea, but @IGJoshua may know of more cases like this one (he was the one who warned me about `(reduce str ...)`).

**Comment by IGJoshua** [comment]
I'm not *currently* aware of any similar cases in clojure core, but I'll admit that even though I'm pretty familiar with the library, I don't know everything about it off the top of my head.

If we really want to hedge our bets on this one then another linter could be like `:prefer-apply` which takes a configurable set of var references for which apply is preferred to reduce and make the default value include `clojure.core/str`.
